[Routing] document mapped route parameters (#20820)
symfony/symfony-docs: doctrine.rst

    The ``message`` option was introduced in Symfony 7.1.

Mapped Route Parameters
~~~~~~~~~~~~~~~~~~~~~~~

When many route parameters are used to find more than one entity,
it is mandatory to use ``#[MapEntity]`` attributes and this can become cumbersome::

>     #[Route('/document/{slug}/{id}-{name}/')]

Reviewer: What do you think about removing the trailing slash from new examples for consistency with current documentation ?

    public function showDocument(
        #[MapEntity(mapping: ['slug' => 'slug'])]
        Category $category,
        #[MapEntity(mapping: ['id' => 'id', 'name' => 'name'])]
        Document $document,
    ): Response
    {

>         // the database queries in this case would be:

Reviewer 2: ```suggestion
        // this would result in the following database queries:
```

Reviewer 2: I think the problem occurs here, when you have `{name}` and `{name}`, right? Name from document and name from category, maybe we should point this out, instead of talking about "cumbersome"?

Author: for information, I copy these text directly from Javier Eguiluz article (https://symfony.com/blog/new-in-symfony-7-1-mapped-route-parameters) to keep an uniformity. (I don't say his text is better than yours ^^)

Author: you are in 7.1 part, problem with name and name is comming next.

        // $document = $documentRepository->findOneBy(['id' => 'the id', 'name' => 'the name']);
        // $category = $categoryRepository->findOneBy(['slug' => 'the slug']);
    }

As an alternative, you can also use Mapped Route Parameters.

> When adding route parameters, you can now define the mapping between the route parameter and the controller argument::

Reviewer 2: ```suggestion
By using mapped route parameters, you can define the mapping between the route parameter and the controller argument::
```
short and precise

    #[Route('/document/{slug:category}/{id:document}-{name:document}')]
    public function showDocument(Document $document, Category $category): Response
    {

>         // the database queries in this case would be:

Reviewer 2: ```suggestion
        // this would result in the following database queries:
```

        // $document = $documentRepository->findOneBy(['id' => 'the id', 'name' => 'the name']);
        // $category = $categoryRepository->findOneBy(['slug' => 'the slug']);
    }

.. versionadded:: 7.1

>     The Mapped Route Parameters was introduced in Symfony 7.1.

Reviewer 2: ```suggestion
    The mapped route parameters were introduced in Symfony 7.1.
```

>     The mapped route parameters were introduced in Symfony 7.1.

Reviewer 2: ```suggestion
    Mapped route parameters were introduced in Symfony 7.1.
```

>     The ``Mapped Route Parameters`` was introduced in Symfony 7.1.

Reviewer: Do we need backticks here as this name is not code, a property's name or something like this ?
